crates/ty_python_semantic/src/types/ide_support.rs

```diff
@@ -278,3 +282,208 @@ pub fn definition_kind_for_name<'db>(
 
     None
 }
+
+/// Label offset information for a parameter within a signature string.
+#[derive(Debug, Clone)]
+pub struct ParameterLabelOffset {
+    /// The start offset of the parameter label within the signature string
+    pub start: usize,
+
+    /// The length of the parameter label
+    pub length: usize,
+}
+
+/// Details about a callable signature for IDE support.
+#[derive(Debug, Clone)]
+pub struct CallSignatureDetails<'db> {
+    /// The signature itself
+    pub signature: Signature<'db>,
+
+    /// The display label for this signature (e.g., "(param1: str, param2: int) -> str")
+    pub label: String,
+
+    /// Label offsets for each parameter in the signature string.
+    /// Each offset specifies the start position and length of a parameter label
+    /// within the full signature string.
+    pub parameter_label_offsets: Vec<ParameterLabelOffset>,
+
+    /// The names of the parameters in the signature, in order.
+    /// This provides easy access to parameter names for documentation lookup.
+    pub parameter_names: Vec<String>,
+
+    /// The definition where this callable was originally defined (useful for
+    /// extracting docstrings).
+    pub definition: Option<Definition<'db>>,
+
+    /// Mapping from argument indices to parameter indices. This helps
+    /// determine which parameter corresponds to which argument position.
+    pub argument_to_parameter_mapping: Vec<Option<usize>>,
```

Comment:
I find this field somewhat unexpected as it isn't specific about a signature, but a particular call to that signature. 

I wonder if we should make this a method instead